packages/shared/src/components/comments/CommentActionButtons.tsx

```diff
@@ -49,9 +50,11 @@ export default function CommentActionButtons({
     setNumUpvotes(comment.numUpvotes);
   }, [comment]);
 
+  const { companionRequest } = useCompanionProtocol();
+  const requestMethod = companionRequest || request;
```

Comment:
Should we not opt to simply pass this as we do with other companion affected request methods?

Reply:
I added this comment before seeing how you use the companion protocol, but see my remarks in the main comment. :)

Reply:
Was also thinking about that too, though I realized I had to drill quite a bit of props to get where the component requires it. So I thought of using this. Thank you!

Reply:
> Like the approach you took on the new hook, although we might have to try this out, it might be super heavy on refocus and add a lot of state that we don't actively use. (Meaning it will never change once you are inside the companion, so why do we use react query for that part?)

I was actually thinking of using Context instead of `useQuery`, thinking it would be quite light still but since you mentioned it, I think we can also go for the Context. 
The only concern I had was we will have to create another ContextProvider since I don't see any of our existing providers where this will fit. And that new context will only have this.
We can also probably declare a static variable just to check if it is the extension then we will use the companion request.
What do you think?

Reply:
Hmm yeah my initial thoughts would be context, but like you mention it would be very small.
Very perhaps we could use it in the bootContext as this now also has a explicit call for bootquery for companion?
Perhaps we can even hook into that and set some setting?

But as mentioned a static variable for the companion would also work for me, for this case.

Reply:
Agreed. Let's go for the static variable then and see how it works out.

Reply:
Now that I worked on it again, I remembered why I was forced to not use static variable - we will have to install `webextension-polyfill-ts` shared package and it is throwing an error if we are trying to load it in an app outside of being an extension, so the webapp always crashes and not load. I think we will have to use Context then.